### ISSUE TITLE
Add support for section document links

### ISSUE THREAD

**Issue by charlespwd** [Shopify/theme-check#243]
Cmd+Click on 'section-name' should drive you to `sections/section-name.liquid` 

```liquid
{% section 'section-name' %}
```

https://github.com/Shopify/theme-check/blob/master/lib/theme_check/language_server/document_link_engine.rb

**Comment by ConduciveMammal** [comment]
Can this also be extended to `include`s? I understand they're deprecated now but still very much exist within the theme world.

**Comment by charlespwd** [comment]
Yeah. It's one thing to not show them as suggestions during code completion to discourage their use, another to provide a worse experience for code that already exists.

Agreed. We'll add it.
